tests/lib/kubernetes/kubernetes_base.py

```diff
@@ -447,12 +447,14 @@ def fetch_kubeconfig(self, destination):
 
 
 class KubernetesBase(ABC):
-    def __init__(self, hardware):
+    def __init__(self, workspace, hardware):
+        self._workspace = workspace
```

Comment:
does that make sense? So hardware and kubernetes can have different workspaces?

Reply:
In theory they could. pytest will provide the same instance of the workspace to both though.

The main thing is that the kubernetes class shouldn't access anything the hardware class creates and vice-versa. If it needs something from the hardware class it should call out to it instead.

We could add some protection around this by giving them unique workspaces or a folder within the workspace, but I think this is sufficient for now.

Reply:
Ah. but still, could we just create a subdir in the hardware workspace for the k8s workspace then?

Reply:
and if we want to have workspace from hardware private, we should mark it as private (self.hardware._workspace)

Reply:
Yes we could create a subdir but I didn't see it necessary yet.

There's a chance a test might want to interact with something that the hardware module has created so being able to access hardware.workspace still makes some sense imo.

Reply:
but then the argument that the kubernetes class shouldn't access anything from hardware is not valid.
I would prefer to not have the workspace parameter in the `__init__` for now and just use the hardware workspace. If we really need it, we can separate it imo.

Reply:
Yes, I agree I'm not being strict with usage here. I find it neater for each module to have access to the workspace, so maybe making them private attributes is better.

I'll leave it like this for the moment while I finish the building of skuba which is part of the reason for the refactor. Then we can revisit.

This probably shouldn't be merge before the CaaSP change anyway.